aws/src/main/java/org/apache/iceberg/aws/AwsClientFactories.java

```diff
@@ -91,13 +92,16 @@ private static AwsClientFactory loadClientFactory(String impl, Map<String, Strin
   static class DefaultAwsClientFactory implements AwsClientFactory {
     private AwsProperties awsProperties;
 
+    private Region region;
+
     DefaultAwsClientFactory() {
       awsProperties = new AwsProperties();
     }
 
     @Override
     public S3Client s3() {
       return S3Client.builder()
+          .region(this.region)
```

Comment:
[Suggestion] Shall we follow the `applyMutation` pattern here
```java
.applyMutation(awsProperties:applyAwsRegionConfiguration)
```
and have the following method in `AwsProperties.java`
```java
public <T extends AwsSyncClientBuilder> void applyAwsRegionConfiguration(T builder) {
   if (awsRegion != null) {
       builder.region(Region.of(awsRegion))
  }
}
```

In this way, we do not need to do any further initialization in the DefaultAwsClientFactories

What do you think about this change?

Reply:
This is good suggestion @JonasJ-ap . I made the change. Please check.